include/neml2/misc/types.h

```diff
@@ -24,88 +24,76 @@
 
 #pragma once
 
-#include <torch/types.h>
-#include <variant>
+#include <c10/util/ArrayRef.h>
+#include <c10/core/TensorOptions.h>
 
-namespace neml2
+namespace at
 {
-using Real = double;
-using Integer = int;
-using Size = int64_t;
-using TensorShape = torch::SmallVector<Size>;
-using TensorShapeRef = torch::IntArrayRef;
-
-// Bring in torch::indexing
+class Tensor;
 namespace indexing
 {
-using namespace torch::indexing;
-using TensorIndices = torch::SmallVector<TensorIndex>;
-using TensorIndicesRef = torch::ArrayRef<TensorIndex>;
+struct Slice;
+struct TensorIndex;
 }
+} // namespace at
 
-/**
- * @brief Traceable size
- *
- * Similar to neml2::TraceableTensorShape, but only for a single dimension.
- * @see neml2::TraceableTensorShape
- */
-struct TraceableSize : public std::variant<Size, torch::Tensor>
+namespace neml2
 {
-  using std::variant<Size, torch::Tensor>::variant;
-
-  /// @return a pointer to the torch::Tensor representing the traceable size if it is traceable, otherwise a nullptr
-  const torch::Tensor * traceable() const noexcept;
-
-  /// @return the concrete size (without any traceable information)
-  Size concrete() const;
+using ATensor = at::Tensor;
+
+/// Fixed width dtypes (mirroring the definition in <torch/csrc/api/include/torch/types.h>)
```

Comment:
What are we gaining by having this huge, static list of aliases?  Also: pretty sure this is the main reason 2.3 no longer works... I'd consider something more portable.

Reply:
We gain compile speed by not including <torch/types.h> (which is pretty dumb and includes almost everything from torch). I've removed the offending device list, so we should be good now.